substrate-node/pallets/pallet-smart-contract/src/lib.rs

```diff
@@ -322,18 +322,20 @@ pub mod pallet {
         SolutionProviderApproved(u64, bool),
         /// A Service contract is created
         ServiceContractCreated(types::ServiceContract),
+        /// A Service contract metadata is set
+        ServiceContractMetadataSet(types::ServiceContract),
+        /// A Service contract fees are set
+        ServiceContractFeesSet(types::ServiceContract),
```

Comment:
No need to emit the full serviceContract here, an ID  and fees would be fine

Reply:
same reason as below
see here https://github.com/threefoldtech/tfchain_graphql/blob/3bbbd797173e018cbf36366e589162952f5eb2ea/src/mappings/serviceContracts.ts#L53